kdc/krb5tgs.c

```diff
@@ -787,7 +787,9 @@ tgs_make_reply(astgs_request_t r,
 	free(et.authorization_data);
     }
     free_LastReq(&ek.last_req);
-    memset(et.key.keyvalue.data, 0, et.key.keyvalue.length);
+    if (et.key.keyvalue.data != NULL) {
+	memset(et.key.keyvalue.data, 0, et.key.keyvalue.length);
```

Comment:
The protection against use of NULL provided that the length is not known to be zero.   
Unrelated, this call should be to memset_s() not memset() which is likely to be optimized away because 'et' is a stack variable going out of scope.